From 80623051136dac17a6d5d844c497cc983125e7a4 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 09 Oct 2022 19:34:01 +0300
Subject: [PATCH] [SW-0] gaudi3: skip pcie rr close config for onix

As PCIe RR space is not available on onix, skip
security_rr_close_pcie_config for onix.

Change-Id: I5d812ddd5bda1f224f4fbb258763938ec6a82af7
---

diff --git a/arch/gaudi3/arc/onix.c b/arch/gaudi3/arc/onix.c
index 7c16e57..a4de718 100644
--- a/arch/gaudi3/arc/onix.c
+++ b/arch/gaudi3/arc/onix.c
@@ -8,6 +8,8 @@
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
 
+void security_rr_close_pcie_config(void) {}
+
 uint64_t arcboot_rom_get_timestamp_freq(void)
 {
 	return SYS_CLOCK_FREQUENCY;
diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 6b3000a..0ea38eb 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -701,7 +701,7 @@
 			arcboot_rom_global.config2_fuse;
 }
 
-void security_rr_close_pcie_config(void)
+__weak void security_rr_close_pcie_config(void)
 {
 	uint32_t pcie_ctrl_val = arcboot_rom_fuse_pcie_ctrl_config();
 
